command/completion/completion.go

```diff
@@ -0,0 +1,107 @@
+package completion
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/smallstep/cli/flags"
+	"github.com/urfave/cli"
+	"go.step.sm/cli-utils/command"
+	"go.step.sm/cli-utils/errs"
+)
+
+func init() {
+	cmd := cli.Command{
+		Name:      "completion",
+		Usage:     "print the shell completion script",
+		UsageText: "**step completion** <shell>",
+		Description: `**step completion** command prints the shell completion script.
+
+## POSITIONAL ARGUMENTS
+
+<shell>
+: The shell program. Either bash or zsh.
+
+## EXAMPLES
+
+Add bash completion for the current user.
+'''
+$ step completion bash >> ~/.bash_completion
+'''
+`,
+		Flags: []cli.Flag{
+			flags.HiddenNoContext,
+		},
+		Action: Completion,
+		BashComplete: func(c *cli.Context) {
```

Comment:
I can't figure out what this method is doing. If I run `step completion` I just get the error which is already in the action about now having enough command line args.

Reply:
This function gets invoked when you write `step completion` and press tab, it will basically check if you haven't already passed an argument, or else output the `bash` and `zsh`. I will test it out again.

Reply:
Oh that's pretty cool! Just tested it out and it works for me.